pkg/netceptor/packetconn.go

```diff
@@ -7,9 +7,25 @@ import (
 	"reflect"
 	"time"
 
+	"github.com/ansible/receptor/pkg/logger"
 	"github.com/ansible/receptor/pkg/utils"
 )
 
+type PacketConner interface {
+	SetHopsToLive(hopsToLive byte)
+	SubscribeUnreachable(doneChan chan struct{}) chan UnreachableNotification
+	ReadFrom(p []byte) (int, net.Addr, error)
+	WriteTo(p []byte, addr net.Addr) (n int, err error)
+	LocalAddr() net.Addr
+	Close() error
+	SetDeadline(t time.Time) error
+	SetReadDeadline(t time.Time) error
+	SetWriteDeadline(t time.Time) error
+	GetCancel() *context.CancelFunc
+	GetLocalService() string
+	GetNetceptorLogger() *logger.ReceptorLogger
+}
+
 // PacketConn implements the net.PacketConn interface via the Receptor network.
 type PacketConn struct {
 	s               *Netceptor
```

Comment:
We can replace this `Netceptor` with an `interface` to make future testing easier.